app/scripts/controllers/network/network-controller.test.ts

```diff
@@ -843,10 +429,19 @@ describe('NetworkController', () => {
                 },
               },
             },
-            async ({ controller, network }) => {
-              network.mockEssentialRpcCalls({
-                latestBlock: POST_1559_BLOCK,
-              });
+            async ({ controller }) => {
```

Comment:
The resulting test code using a fake provider instead of Nock is slightly more verbose because of the API that I had built over on the `core` side. I could probably make this a little simpler, but it could also be argued that seeing things a bit more spelled out and knowing which RPC methods that NetworkController calls under the hood instead of making some things implicit is easier to read.